test/cmd/newapp.sh

```diff
@@ -53,14 +53,13 @@ os::cmd::expect_success 'oc delete is myruby'
 os::cmd::expect_failure_and_text 'oc new-app https://github.com/openshift/nodejs-ex --strategy=docker' 'No Dockerfile was found'
 
 # repo related error message validation
-os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'mysql-persisten as a local directory'
-os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'mysql as a local directory'
-os::cmd::expect_failure_and_text 'oc new-app --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-app https://www.google.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-app https://examplegit.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build https://www.google.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build https://examplegit.com/openshift/nodejs-e' 'as a Git repository URL:  '
+os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'only a partial match was found'
+os::cmd::expect_failure_and_text 'oc new-app --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'only a partial match was found'
```

Comment:
I wouldn't expect this change...  feels like a step backwards from the current output:

```
$ oc new-app --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git
error: Errors occurred while determining argument types:

https://192.30.253.113/openshift/ruby-hello-world.git as a Git repository URL:  fatal: unable to access 'https://192.30.253.113/openshift/ruby-hello-world.git/': SSL: no alternative certificate subject name matches target host name '192.30.253.113'

https://192.30.253.113/openshift/ruby-hello-world.git as a local directory pointing to a Git repository:  stat https://192.30.253.113/openshift/ruby-hello-world.git: no such file or directory

Errors occurred during resource creation:
error: --strategy is specified and none of the arguments provided could be classified as a source code location
```

Reply:
(If i were to change anything about the above output, i'd skip the resource creation error output and just abort after the argument determination errors)

Reply:
this change centers around my observation about the component ref classification rarely failing

I had been consider shifting my check from the classification logic (though it was more elegant) to the handleError logic ... your concern has prompted me to do that 

I believe that will result in reverting the change here

Reply:
One more point as a result of the shift though ... I'll be curious after I start posting some sample output if you'll want to qualify *on which* runtime errors the classification errors get posted.

I wonder if we'll want to reconcile the type of error noted in functions like https://github.com/openshift/origin/blob/master/pkg/oc/cli/cmd/newapp.go#L752-L878 with the output you want to see with say the invocation from @GrahamDumpleton that stemmed this issue.

I'll provide some precise examples in a bit if that helps the determination.

Reply:
OK, I have a set of changes on my laptop, that if we go with, would result in:

1) only changing the newapp.sh test to account for the @bparees requested transition of `no match for` to `unable to locate resource for`

2) output of the following for @GrahamDumpleton 's example:

```
error: unable to locate resource for "xxx"

The 'oc new-build' command will match arguments to the following types:

  1. Images tagged into image streams in the current project or the 'openshift' project
     - if you don't specify a tag, we'll add ':latest'
  2. Images in the Docker Hub, on remote registries, or on the local Docker engine
  3. Git repository URLs or local paths that point to Git repositories

--allow-missing-images can be used to force the use of an image that was not matched

See 'oc new-build -h' for examples.
For further assistance,  new-build classification of argument types resulted in the following:

python~https://github.com/openshift-katacoda/blog-django-py as a Git repository URL:  fatal: I don't handle protocol 'python~https'

python~https://github.com/openshift-katacoda/blog-django-py as a local directory pointing to a Git repository:  stat python~https://github.com/openshift-katacoda/blog-django-py: no such file or directory

The argument python~https://github.com/openshift-katacoda/blog-django-py passed the component classification tests
```

Thoughts ?